weather_dl/download_pipeline/parsers.py

```diff
@@ -149,8 +149,10 @@ def typecast(key: str, value: t.Any) -> t.Any:
 def parse_config(file: t.IO) -> t.Dict:
     """Parses a `*.json` or `*.cfg` file into a configuration dictionary."""
     try:
-        # TODO(b/175429166): JSON files do not support MARs range syntax.
-        return json.load(file)
+        config = json.load(file)
+        config_by_section = {s: _parse_lists(v, s) for s, v in config.items()}
+        config_with_nesting = parse_subsections(config_by_section)
+        return config_with_nesting
```

Comment:
This is great! 

Is there some way we can combine the logic of the JSON and Config parsing? It seems like there are repeated steps here.

Reply:
Yes.
Just needed to add one condition for it _if not config:_